include/ua_plugin_securitypolicy.h

```diff
@@ -0,0 +1,424 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef UA_PLUGIN_SECURITYPOLICY_H_
+#define UA_PLUGIN_SECURITYPOLICY_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "ua_types.h"
+#include "ua_log.h"
+#include "ua_types_generated.h"
+
+extern const UA_ByteString UA_SECURITY_POLICY_NONE_URI;
+
+//////////////////////////
+// Forward declarations //
+//////////////////////////
+struct UA_SecurityPolicy;
+typedef struct UA_SecurityPolicy UA_SecurityPolicy;
+
+struct UA_Channel_SecurityContext;
+typedef struct UA_Channel_SecurityContext UA_Channel_SecurityContext;
+
+struct UA_Endpoint_SecurityContext;
+typedef struct UA_Endpoint_SecurityContext UA_Endpoint_SecurityContext;
+/////////////////////////////////
+// End of forward declarations //
+/////////////////////////////////
+
+/**
+ * This module contains the signing algorithms
+ * used by a security policy.
+ */
+typedef struct
+{
+    /**
+     * Verifies the signature of the message using the provided certificate.
+     *
+     * \param message the message to verify.
+     * \param context the context that contains the key to verify the supplied message with.
+     */
+    UA_StatusCode (*const verify)(const UA_SecurityPolicy *securityPolicy,
+                                  const void *context,

```

Comment:
Since the module can be used for either symmetric verification or asymmetric verification, this can be both. It depends on which module we call. That's the price we pay for having only one declaration for both versions.